invenio/b2share/modules/b2deposit/restful.py

```diff
@@ -205,9 +206,14 @@ def get(self, oauth, domain_name, **kwargs):
         # get domain id from domain name
         domain = Bib98x.query.filter_by(value=domain_name).first()
         if domain is None:
-            abort(404, status=404,
-                  message="Please try a valid domain name: " +
-                         ", ".join(metadata_classes().keys()))
+            if domain_name not in metadata_classes().keys():
```

Comment:
This check should be moved up, before line 206 (before asking the database). It is part of the initial checks of the function, right?